### DISCUSSION TITLE
Is there a basic example available?

### DISCUSSION THREAD

**Discussion by drkdw** [TukuToi/better-wp-plugin-boilerplate#16] [General]
Hi there!

I'm wondering if there is a basic example available of this plugin with a few option just to understand where everything goes. I'm new to the boilerplate/OOP and it's pretty intimidating. A proper example would be a great help!

Looking forward to your comments,

**Comment by smileBeda** [comment]
Hello, please apologise the late reply. I do not get notices when someone asks questions here. I'll fix that, but meanwhile let me reply to your question:
yes, the basic example _is_ the plugin boilerplate

This folder here represents a fully functional Plugin that even registers one shortcode https://github.com/TukuToi/better-wp-plugin-boilerplate/tree/master/plugin-name
You can activate, deactivate and delete it in the WP or CP Admin.

You can see several approaches of using this boilerplate in _all_ these repos https://github.com/orgs/TukuToi/repositories (from simple to very complex), or these too
https://github.com/smileBeda/tukutoi-maintenance
https://github.com/smileBeda/tukutoi-contact-form

I know the OOP approach can be confusing. Massively, actually. And this plugin is nowhere _perfect_.
Many things could be done better. I suggest to start smaller, if you want to start learning OOP.
There are many tutorials and examples on the net googling for "OOP PHP Basics"
And immediately you will get confused because everyone will be talking about "car implements model" :rofl:
I have _not_ yet found a good example of OOP that is "graspable" without making weird comparisons.

The best is, start very small, see what the code does and build from there.
The Answer on this SO thread (just above my linked comment) is a perhaps more relatable view to OOP https://stackoverflow.com/questions/1197616/can-you-help-me-understand-php-classes-a-little-better/1197679#comment127173455_1197679

But really, it is a journey that never ends. 

From "A class is just a group of functions", the sky is the limit.
